scripts/smb-enum-services.nse

```diff
@@ -0,0 +1,1144 @@
+local msrpc = require "msrpc"
+local smb = require "smb"
+local smbauth = require "smbauth"
+local stdnse = require "stdnse"
+local string = require "string"
+local table = require "table"
+local http = require "http"
+local shortport = require "shortport"
+
+description = [[
+Retrieves the list of services running on a remote Windows system.
+Each service attribute contains service name, display name and service status of
+each service.
+
+References:
+* https://msdn.microsoft.com/en-us/library/windows/desktop/ms682637(v=vs.85).aspx
+* https://msdn.microsoft.com/en-us/library/windows/desktop/ms682651(v=vs.85).aspx
+* https://github.com/samba-team/samba/blob/d8a5565ae647352d11d622bd4e73ff4568678a7c/librpc/idl/svcctl.idl
+]]
+
+-- @usage
+-- nmap --script smb-enum-services.nse -p445 <host>
+-- nmap --script smb-enum-services.nse --script-args smbusername=<username>,smbpass=<password> -p445 <host>
+--
+-- The following lines displays the normal and xml results when this script
+-- was run against Windows 2003 R2 x64 Enterprise Server.
+--
+-- @output
+-- | smb-enum-services:
+-- |
+-- |     displayName: Application Layer Gateway Service
+-- |     serviceName: ALG
+-- |     serviceStatus:
+-- |       check_point: 0
+-- |       wait_hint: 0
+-- |       state:
+-- |
+-- |       win32_exit_code: 0
+-- |       service_exit_code: 0
+-- |       type:
+-- |         SERVICE_TYPE_WIN32
+-- |         SERVICE_TYPE_WIN32_OWN_PROCESS
+-- |       controls_accepted:
+-- |         SERVICE_CONTROL_NETBINDADD
+-- |         SERVICE_CONTROL_CONTINUE
+-- |         SERVICE_CONTROL_STOP
+-- |         SERVICE_CONTROL_NETBINDENABLE
+-- |
+-- |     displayName: ClipBook
+-- |     serviceName: ClipSrv
+-- |     serviceStatus:
+-- |       check_point: 0
+-- |       wait_hint: 0
```

Comment:
`check_point`, `wait_hint`, `win32_exit_code`, and `service_exit_code` are not really helpful or interesting items, so in the interest of reducing the size of the output somewhat, let's leave them out. Loop over the return table of `svcctl_enumservicesstatusw` and create a `stdnse.output_table` for each one, filling it out with the values we care about. This will ensure that the keys are in the same order for each service, which is important in reading and comparing results.